c/src/ml-api-common.c

```diff
@@ -1033,6 +1085,24 @@ ml_tensors_data_set_tensor_data (ml_tensors_data_h data, unsigned int index,
     goto report;
   }
 
+  /**
+   * By default, the tensor format is _NNS_TENSOR_FORMAT_STATIC.
+   * In this case, memory allocation and the setting of _data->tensors[index].size
+   * are already handled in ml_tensors_data_create().
+   * So for the STATIC format, both the `size` and `data` pointer should already be valid here.
+   *
+   * For FLEXIBLE format, memory may not be allocated yet and will be handled here.
+   */
+  _info = (ml_tensors_info_s *) _data->info;
+  if (_info && _info->info.format == _NNS_TENSOR_FORMAT_FLEXIBLE) {
+    _ml_logw
```

Comment:
We cannot assure memory is not allocated here, user may call this function twice with different buffer.
```
ml_tensors_data_create ();
ml_tensors_data_set_tensor_data (data, 0, buffer1, size);
ml_tensors_data_set_tensor_data (data, 0, buffer2, size); // mem leak case

```

Reply:
Thank you for taking the time to review.
I've updated it. Thank you.